REST API updates (#226)
LucidtechAI/cradl-docs: docs/reference/restapi.md

## Changelog

### 2024-12-02

- Added `async` to `POST /predictions`. Async predictions are now supported. 

> - Added `error` and `status` to `POST /predictions` output.
> - Changed `inferenceTime` and `predictions` in `POST /predictions` output, they are now optional.

Reviewer: ```suggestion
- Added `error` and `status` to `/predictions`.
- Changed `inferenceTime` and `predictions` in `/predictions` to be optional.
```
